### DISCUSSION TITLE
Prisma `Upsert` works in `Update` but not `Create`

### DISCUSSION THREAD

**Discussion by ACPK** [prisma/prisma#11966] [Q&A]
### Bug description

Does Prisma allow for upsert inside of a create?  I am able to put an upsert inside of an update [example from Prisma Docs](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#update-or-create-a-related-record_), but am not able to have an upsert inside of a create.

Error:

```
Unknown arg `upsert` in data.user.upsert for type usersCreateNestedOneWithoutConversationsInput. Did you mean `select`? Available args:
type usersCreateNestedOneWithoutConversationsInput {
  create?: usersCreateWithoutConversationsInput | usersUncheckedCreateWithoutConversationsInput
  connectOrCreate?: usersCreateOrConnectWithoutConversationsInput
  connect?: usersWhereUniqueInput
```

Query I am trying to run:
```
const createConversation = await prisma.conversations.create({
    data: {
    body: 'This is a test',
    user: {
      upsert: {
        where: {
            email: 'ak@prisma.io',
        },
        create: {
            email: 'ak@prisma.io',
        },
        update: {},
      },
    },
  },
})
```

**Comment by ACPK** [comment]
From the docs, it looks like I should be able to use upsert inside of a `create`.  It's fine if this feature isn't built as I want to make sure I'm not doing anything wrong.

An example of the working `update` is below.

```
const createConversation = await prisma.conversations.update({
  where: {
    id: 1,
  },
  data: {
    body: 'This is a test',
    user: {
      upsert: {
        create: {
          email: 'bob@prisma.io',
          name: 'Bob the New User',
        },
        update: {
          email: 'bob@prisma.io',
          name: 'Bob the existing user',
        },
      },
    },
  },
})
```

**Reply by austincrim** [reply]
Hey @ACPK 👋 ,

Could you share your Prisma schema so I can recreate this locally?

**Reply by vladi-stevanovic** [reply]
👋 Hello @ACPK  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Reply by k0l11** [reply]
Hi, I'm also facing this problem. I want to be able to do an upsert inside of a nested write when creating a record `D`. `D` is deep down the hierarchy: `A` can have multiple `B`, `B` can have multiple `C`, `C` can have multiple `D` (it's 1:n for all of them). When `D` is being created, it's possible that `A`, `B` or `C` don't exist yet. The `connectOrCreate` API covers the creation and connecting, but it does not cover updating `A`, `B` and `C` in case they exist, but there's new data for them when `D` is being created. All of this should happen inside of one transaction. I could split this into one `create` using `connectOrCreate` and one nested `update` with both of them getting wrapped in a `$transaction`, but I feel like being able to use `upsert` in `create` would be a cleaner solution here.
Related: https://github.com/prisma/prisma/issues/3538

**Reply by Luigi-Sirius** [reply]
I'm facing the same problem!
Quite annoying and misleading, is there a solution for this? or at least planed? 

Thanks

**Reply by bcanfield** [reply]
Also spent a little too much time dealing with this issue.

My use case is that instead of connecting or creating, I want to update or create.

**Comment by janpio** [comment]
Hey, if something does not work please make sure to create a bug report issue. The engineering team can not triage discussions as input for our development. Thanks.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
